src/App.tsx

```diff
@@ -1,12 +1,78 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { GoodList } from './components';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+enum Request {
+  all,
+  first5,
+  red,
+}
+
+const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+
+  const handleClick = useCallback(async (request: Request) => {
+    switch (request) {
+      case Request.first5: {
+        const data = await get5First();
+
+        setGoods(data);
+      }
+
+        break;
+
+      case Request.red: {
+        const data = await getRedGoods();
+
+        setGoods(data);
+      }
+
+        break;
+
+      default: {
+        const data = await getAll();
+
+        setGoods(data);
+      }
+
+        break;
+    }
+  }, []);
+
+  return (
+    <div className="app">
+      <h1>Dynamic list of Goods</h1>
+      <div className="app__buttons">
+        <button
+          type="button"
+          onClick={() => handleClick(Request.all)}
```

Comment:
you can expect a callback on handleClick and use the result of calling it

```suggestion
          onClick={() => handleClick(get5First)}
```

Reply:
🤦‍♂️ it's hell logical